File: laravel/app/Models/Reply.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Reply extends Model
{
    use SoftDeletes;

    protected $fillable = ['post_id', 'user_id', 'parent_reply_id', 'body'];

    public function post(): BelongsTo       { return $this->belongsTo(Post::class); }
    public function author(): BelongsTo     { return $this->belongsTo(User::class, 'user_id'); }
    public function parent(): BelongsTo     { return $this->belongsTo(Reply::class, 'parent_reply_id'); }
    public function children(): HasMany     { return $this->hasMany(Reply::class, 'parent_reply_id'); }
}

File: java-gui/src/main/java/com/smartforum/model/Reply.java
package com.smartforum.model;

/** Mirrors the Laravel Reply model (replies table). */
public class Reply {
    public final int    id;
    public final int    postId;
    public final int    userId;
    public final String authorName;
    public final String body;
    public final String createdAt;

    public Reply(int id, int postId, int userId, String authorName, String body, String createdAt) {
        this.id         = id;
        this.postId     = postId;
        this.userId     = userId;
        this.authorName = authorName;
        this.body       = body;
        this.createdAt  = createdAt;
    }
}
